x-pack/plugins/alerting/common/rule_snooze_type.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export type RuleSnooze = Array<{
```

Comment:
> Similar to this, it seems like we still have `muteAll` around, to handle the case of "mute forever". 

I tried merely changing the name of `muteAll` and it required changes in like 80 files. I agree we could probably remove it but not before feature freeze. This PR is huge enough already.

> Also not clear, but I don't think we're ever actually setting rRule right now?

No, this PR just changes the data model, not the APIs or the UIs. Setting the rRule comes when we actually implement the new recurring snooze functionality on the frontend.

> It seems like we could actually just create an rRule in this case, so wouldn't have to treat the snooze case as special with the startTime, and could then get rid of startTime completely. That way we'd always have the case where a snoozeSchedule entry would have a non-null rRule.

Possible, though having a `null` rRule is a quick and easy way to determine if a snooze is or is not recurring, without having to run any parsing on the rRule. I suppose re-implementing the `rrule` lib structure in the savedobject instead of using an rRule string could give us that for free, though, and just use `count: 1` as an indicator of a non-recurring snooze. Again, though, feature freeze is next week and we don't even have an API that sets scheduled and recurring snoozes yet.

Reply:
Tracking these changes in https://github.com/elastic/kibana/issues/132266

Reply:
Actually managed to implement moving `startTime` into the `rRule`, and requiring it for all snoozes. Still need to kick removing `muteAll` down the road to #132266 though.